src/Models/Context/SceneThemeContext/SceneThemeContext.tsx

```diff
@@ -43,10 +44,9 @@ export const SceneThemeContextReducer: (
         );
         switch (action.type) {
             case SceneThemeContextActionType.SET_OBJECT_COLOR: {
-                draft.objectColor =
-                    action.payload.color ||
-                    draft.objectColorOptions?.[0]?.color ||
-                    '';
+                draft.objectColor = draft.objectColorOptions.find(
+                    (x) => x.color === action.payload.color
```

Comment:
should there be a fallback option here in case the color is not found in the payload?

Reply:
hmm. I mean ideally yeah. That only really comes into play if consumers customize and somehow botch it. I could fallback to the default color object I guess.

Reply:
added